delira/training/experiment.py

```diff
@@ -1222,3 +1222,140 @@ def test(self, network, test_data: BaseDataManager,
+                             key_mapping=key_mapping,
+                             val_score_key=val_score_key,
+                             optim_builder=optim_builder,
+                             checkpoint_freq=checkpoint_freq,
+                             trainer_cls=trainer_cls,
+                             **kwargs)
+
+        def test(self, network: AbstractChainerNetwork,
+                 test_data: BaseDataManager,
+                 metrics: dict, metric_keys=None,
+                 verbose=False, prepare_batch=None,
+                 convert_fn=None, **kwargs):
+            """
+            Setup and run testing on a given network
+
+            Parameters
+            ----------
+            network : :class:`AbstractNetwork`
+                the (trained) network to test
+            test_data : :class:`BaseDataManager`
+                the data to use for testing
+            metrics : dict
+                the metrics to calculate
+            metric_keys : dict of tuples
+                the batch_dict keys to use for each metric to calculate.
+                Should contain a value for each key in ``metrics``.
+                If no values are given for a key, per default ``pred`` and
+                ``label``
+                 will be used for metric calculation
+            verbose : bool
+                verbosity of the test process
+            prepare_batch : function
+                function to convert a batch-dict to a format accepted by the
+                model. This conversion typically includes dtype-conversion,
+                reshaping, wrapping to backend-specific tensors and
+                pushing to correct devices. If not further specified uses the
+                ``network``'s ``prepare_batch`` with CPU devices
+            convert_fn : function
+                function to convert a batch of tensors to numpy
+                if not specified defaults to :func:`convert_torch_tensor_to_npy`
+            **kwargs :
+                additional keyword arguments
+
+            Returns
+            -------
+            dict
+                all predictions obtained by feeding the ``test_data`` through
+                the ``network``
+            dict
+                all metrics calculated upon the ``test_data`` and the obtained
+                predictions
+
+            """
+
+            # use backend-specific and model-specific prepare_batch fn
+            # (runs on same device as passed network per default)
+
+            device = network.device
+            if prepare_batch is None:
+                prepare_batch = partial(network.prepare_batch,
+                                        input_device=device,
+                                        output_device=device)
+
+            # switch to backend-specific convert function
+            if convert_fn is None:
+                convert_fn = convert_chainer_tensor_to_npy
```

Comment:
Why isn't this added as a default argument to the function?

Reply:
Will change this